rpcs3/main.cpp

```diff
@@ -494,27 +496,37 @@ int main(int argc, char** argv)
 	{
 		sys_log.notice("Booting application from command line: %s", args.at(0).toStdString());
 
+		auto bootRom = sstr(args.at(0));
+
 		// Propagate command line arguments
 		std::vector<std::string> argv;
 
-		if (args.length() > 1)
+		// we skip the first entry (which should be the application eboot.bin etc)
+		for (int i = 1; i < args.length(); i++)
 		{
-			argv.emplace_back();
-
-			for (int i = 1; i < args.length(); i++)
-			{
-				const std::string arg = args[i].toStdString();
-				argv.emplace_back(arg);
-				sys_log.notice("Optional command line argument %d: %s", i, arg);
-			}
+			const std::string arg = args[i].toStdString();
+			argv.emplace_back(arg);
+			sys_log.notice("Optional command line argument %d: %s", i, arg);
 		}
 
 		// Ugly workaround
-		QTimer::singleShot(2, [config_override_path, path = sstr(QFileInfo(args.at(0)).absoluteFilePath()), argv = std::move(argv)]() mutable
+		QTimer::singleShot(2, [config_override_path, bootRom, argv=std::move(argv), &app]() mutable
 		{
 			Emu.argv = std::move(argv);
 			Emu.SetForceBoot(true);
-			Emu.BootGame(path, "", true);
+			if (bootRom[0] == '[')
+			{
+				// we'll boot from title only
+				bootRom = bootRom.substr(1, bootRom.find("]")-1);
+				Emu.BootGame("", bootRom, true);
+				app->exit();
+			}
+			else
+			{
+				bootRom = sstr(QFileInfo(QString::fromStdString(bootRom)).absoluteFilePath());
+				Emu.BootGame(bootRom, "", true);
+				app->exit();
```

Comment:
don't exit.... wtf

Reply:
Ever? or in line with my commit message, where it should perhaps be only performed in certain situations (like in no-gui / headless, or if an --exit-after parameter is met).

Reply:
The code makes zero sense.
And we already have a config entry for that anyway

Reply:
What's "zero sense" about it?  If there's an actual technical problem, by all means, share the details.

It shouldn't be required to write an entire configuration file for just this.
I also suspect the configuration file option doesn't work if the Emu.BootGameXxx() option doesn't succeed.